.dotnet/src/Custom/Chat/ChatCompletionOptions.cs

```diff
@@ -144,6 +145,32 @@ public ChatCompletionOptions()
     [Experimental("OPENAI001")]
     public long? Seed { get; set; }
 
+    // CUSTOM: Hide deprecated max_tokens and reroute to newer max_completion_tokens.
+    [CodeGenMember("MaxTokens")]
+    internal int? _deprecatedMaxTokens { get; set; }
+
+    // CUSTOM: Add legacy `max_tokens` routing.
+    /// <summary>
+    /// An upper bound for the number of tokens that can be generated for a completion, including visible output tokens and, on applicable models, reasoning tokens.
+    /// </summary>
+    public int? MaxTokens
```

Comment:
I wonder if we should rename this property to `MaxOutputTokens` or `MaxOutputTokenCount`. This would make it clear that this value only affects output tokens and ignores input tokens entirely. Anecdotally, I wasn't sure myself until I read the documentation.

Reply:
I like the consistency. May as well do all the renames right now -- this is the time!